Assets/Script/Patterns/Tank.cs

```diff
@@ -0,0 +1,52 @@
+using UnityEngine;
+
+public class Tank : MonoBehaviour, ITank
+{
+
+    public MeshFilter bodyMesh;
+    public MeshFilter cannonMesh;
+    public MeshFilter turretMesh;
+    public MeshFilter wheelMesh;
+
+    [HideInInspector] public GameObject projectile;
```

Comment:
Prefer to use properties in these cases.

`public GameObject Projectile {get; set;}`

In C# it is always preferrable to hide fields and use properties: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1051